docs/en/observability/monitor-k8s/monitor-k8s-application-performance.asciidoc

```diff
@@ -0,0 +1,304 @@
+[discrete]
+[[monitor-kubernetes-application-performance]]
+== Part 3: Monitor application performance
+
+Quickly triage and troubleshoot application performance problems with the help of Elastic
+application performance monitoring (APM).
+
+Think of a latency spike -- APM can help you narrow the scope of your investigation to a single service.
+Because you've also ingested and correlated logs and metrics, you can then link the problem to CPU and memory utilization via the error log entries of a particular Kubernetes pod.
+
+[discrete]
+=== Step 1: Set up APM Server
+
+Application monitoring data is streamed from your applications running in Kubernetes to APM Server,
+where it is validated, processed, and transformed into {es} documents.
+
+There are many ways to deploy APM Server when working with Kubernetes.
+
+This guide assumes you're using our hosted {ess} on {ecloud}.
+If you want to manage APM Server yourself, there are a few alternative options:
+
+* {eck-ref}/[Elastic Cloud on Kubernetes (ECK)] -- The Elastic recommended approach for managing
+APM Server deployed with Kubernetes.
+Built on the Kubernetes Operator pattern, ECK extends basic Kubernetes orchestration capabilities
+to support the setup and management of APM Server on Kubernetes.
+
+* Deploy APM Server as a DaemonSet -- Ensure a running instance of APM Server on each node in your cluster.
+Useful when all pods in a node should share a single APM Server instance.
+
+* Deploy APM Server as a sidecar -- For environments that should not share an APM Server,
+like when directing traces from multiple applications to separate {es} clusters.
+
+* {apm-server-ref-v}/installing.html[Download and install APM Server] -- The classic, non-Kubernetes option.
+
+[discrete]
+=== Step 2: Set up APM Agents
+
+In most cases, setting up APM agents and thereby instrumenting your applications
+is as easy as installing a library and adding a few lines of code.
+See the relevant agent setup instructions for more information:
+
+* {apm-go-ref-v}/getting-started.html[Go agent]
+* {apm-java-ref-v}/setup.html[Java agent]
+* {apm-dotnet-ref-v}/setup.html[.NET agent]
+* {apm-node-ref-v}/set-up.html[Node.js agent]
+* {apm-py-ref-v}/set-up.html[Python agent]
+* {apm-ruby-ref-v}/set-up.html[Ruby agent]
+
+[discrete]
+=== Step 3: (Optional) Attach the agent
+
+Some agents offer the ability to instrument supported technologies without
+any changes to an application image or code.
+To do this, you'll need an https://kubernetes.io/docs/concepts/workloads/pods/init-containers/[init container]
+based on an official Elastic APM docker image.
+
+Before your application starts, copy the agent from the init container into a shared volume.
+For example, with the Java agent:
+
+[source,yml]
+----
+    # ...
+    spec:
+      volumes:
+      - name: elastic-apm-agent <1>
+        emptyDir: {}
+      initContainers:
+      - name: elastic-java-agent
+        image: docker.elastic.co/observability/apm-agent-java:1.12.0 <2>
```

Comment:
I am not sure, but I think 1.12.0 is old.  Is there a way to sync the docs with the current release number of the agent?

Reply:
Not yet, unfortunately. I'll have to update it manually.

Reply:
Updated to `1.19.0` in 10a4ea8.